packages/graphql/src/classes/Subgraph.ts

```diff
@@ -0,0 +1,291 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { buildSubgraphSchema } from "@apollo/subgraph";
+import { mergeTypeDefs } from "@graphql-tools/merge";
+import { IResolvers, makeDirectiveNode, TypeSource } from "@graphql-tools/utils";
+import {
+    ConstDirectiveNode,
+    DefinitionNode,
+    DocumentNode,
+    GraphQLDirective,
+    GraphQLNamedType,
+    GraphQLResolveInfo,
+    Kind,
+    parse,
+    print,
+    SchemaExtensionNode,
+} from "graphql";
+import type * as neo4j from "neo4j-driver";
+import { translateResolveReference } from "../translate/translate-resolve-reference";
+import type { Context, Node } from "../types";
+import { execute } from "../utils";
+import getNeo4jResolveTree from "../utils/get-neo4j-resolve-tree";
+import { Executor } from "./Executor";
+
+type FederationDirectiveName =
+    | "key"
+    | "extends"
+    | "shareable"
+    | "inaccessible"
+    | "override"
+    | "external"
+    | "provides"
+    | "requires"
+    | "tag";
+
+type FullyQualifiedFederationDirectiveName = `federation__${FederationDirectiveName}`;
+
+const isFederationDirectiveName = (name: string): name is FederationDirectiveName =>
+    ["key", "extends", "shareable", "inaccessible", "override", "external", "provides", "requires", "tag"].includes(
+        name
+    );
+
+export class Subgraph {
+    private importArgument: Map<
+        FederationDirectiveName,
+        FederationDirectiveName | FullyQualifiedFederationDirectiveName | string
+    >;
+    private typeDefs: TypeSource;
+    private linkExtension: SchemaExtensionNode;
+
+    constructor(typeDefs: TypeSource) {
+        this.typeDefs = typeDefs;
+
+        this.importArgument = new Map([
+            ["key", "federation__key"],
+            ["extends", "federation__extends"],
+            ["shareable", "federation__shareable"],
+            ["inaccessible", "federation__inaccessible"],
+            ["override", "federation__override"],
+            ["external", "federation__external"],
+            ["provides", "federation__provides"],
+            ["requires", "federation__requires"],
+            ["tag", "federation__tag"],
+        ]);
+
+        const linkMeta = this.findFederationLinkMeta(typeDefs);
+        if (!linkMeta) {
+            throw new Error(`typeDefs must contain \`@link\` schema extension to be used with Apollo Federation`);
+        }
+        const { extension, directive: linkDirective } = linkMeta;
+
+        this.linkExtension = extension;
+
+        this.parseLinkImportArgument(linkDirective);
+    }
+
+    public getFullyQualifiedDirectiveName(
+        name: FederationDirectiveName
+    ): FederationDirectiveName | FullyQualifiedFederationDirectiveName | string {
```

Comment:
I think this type is irrelevant here, because  both `FederationDirectiveName` and `FullyQualifiedFederationDirectiveName` are subtypes of `string`, this type is going to be equal to `string`